### PR TITLE
Implement comment functionality with 'ok' UI

### PR DESCRIPTION
**Pull request by NicolasCapparelli**
- Add dependencies to pom.xml
- Implement Connect.vue
- Implement Servlet code

- Ignore the stuff under main/webapp, these are distribution files that get auto-updated when I build the project, and are responsible for such a huge number of additions on these commits

### CODE REVIEW

File: portfolio/src/components/Connect.vue [hunk 1]
```diff
@@ -1,15 +1,116 @@
...
+        </div>
+
+        <h1 style="width: 100%;">Leave a Comment</h1>
+        <div id="comments-container">
+            <div id="message-input-container" class="comment">
+                <v-text-field 
+                    style="width: 100%" 
+                    label="Name"
+                    v-model="name"
+                    color="#DBDBDB"
```
Comment (shuangjiec) [inline portfolio/src/components/Connect.vue]: this seems like a value used many times. Are you able to create a constant for this value with readable name e.g. main_text_color or etc and reference it here?

File: portfolio/src/components/Connect.vue [hunk 2]
```diff
@@ -1,15 +1,116 @@
...
+            await this.getDataFromServer()
+
+        },
+
+        getDataFromServer: async function () {
+            let response = await fetch('https://8080-dot-12454009-dot-devshell.appspot.com/comments')
+
+            if (response.ok) {
+                let respData = await response.json()
+                console.log(respData)
```
Comment (shuangjiec) [inline portfolio/src/components/Connect.vue]: Is it for debug? ideally in real life, we don't want to print these data in the console?

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  public Comment(String name, String message,  String timestamp, Long id) {
+    this.name = name;
+    this.message = message;    
+    this.timeCreated = timestamp;
+    this.id = id;
+  }
+
+  public void setDate(String newDate) {
+    this.timeCreated = newDate;
+  }
```
Comment (shuangjiec) [inline portfolio/src/main/java/com/google/sps/data/Comment.java]: Why is this block necessary

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+public final class Comment {
+  String name;
+  String message;
+  String timeCreated;
+  Long id;
+
+  public Comment(String name, String message,  String timestamp, Long id) {
+    this.name = name;
+    this.message = message;    
+    this.timeCreated = timestamp;
```
Comment (shuangjiec) [inline portfolio/src/main/java/com/google/sps/data/Comment.java]: I would name it createDate or similar. timstamp usually means the epoch time. usually, for website, server saves epoch time when the request commited and then client convert epoch time to whatever format they want.

File: portfolio/src/components/Connect.vue (hunk 2 above)
Comment (NicolasCapparelli) [reply portfolio/src/components/Connect.vue]: Yep, I was having this weird issue where the data was coming back in a different format and forgot to remove, thank you!

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 3 above)
Comment (NicolasCapparelli) [reply portfolio/src/main/java/com/google/sps/data/Comment.java]: In an earlier implementation I was setting the date via this setter, but I found out I can use gson to directly convert response -> Comment object. Forgot to remove this after new implementation, thanks! Will resolve in next commit.

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 4 above)
Comment (NicolasCapparelli) [reply portfolio/src/main/java/com/google/sps/data/Comment.java]: Ahh ok, I will create epoch timestamp on the server and, as you mentioned, format on the client, thank you!

File: portfolio/src/components/Connect.vue (hunk 1 above)
Comment (NicolasCapparelli) [reply portfolio/src/components/Connect.vue]: Ok, made a constants file that exports a COLORS. Will be in next commit

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -15,18 +15,78 @@
...
+
+        // Create JSON object with GSON
+        String requestData = request.getReader().lines().collect(Collectors.joining());
+        Comment comment = new Gson().fromJson(requestData, Comment.class);
+
+        // Creating DataStore Entity
+        Entity taskEntity = new Entity("Comment");
+        taskEntity.setProperty("name", comment.getName());
+        taskEntity.setProperty("message", comment.getMessage());
+        taskEntity.setProperty("timestamp", timestamp);
```
Comment (shuangjiec) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:84]: create constants for these strings so we know there won't be typo when getting/setting values.

**Review by shuangjiec — APPROVED** [review]

**Review by shuangjiec — APPROVED** [review]

**Pull request merged**